common/src/gas_provider/node.rs

```diff
@@ -20,14 +20,30 @@ use super::*;
 use core::ops::{Add, Index, IndexMut};
 use enum_iterator::cardinality;
 use gear_core::ids::ReservationId;
+use scale_decode::DecodeAsType;
+use scale_encode::EncodeAsType;
 use sp_runtime::{
     codec::{self, MaxEncodedLen},
     scale_info,
     traits::Zero,
 };
 
 /// ID of the [`GasNode`].
-#[derive(Debug, Copy, Clone, Hash, Eq, PartialEq, Ord, PartialOrd, Encode, Decode, TypeInfo)]
+#[derive(
+    Debug,
+    Copy,
+    Clone,
+    Hash,
+    Eq,
+    PartialEq,
+    Ord,
+    PartialOrd,
+    Encode,
+    EncodeAsType,
```

Comment:
Exactly this type doesn't require `EncodeAsType` and `DecodeAsType`. Are they for the future PRs?

Reply:
I've removed these changes from this PR